Gems/Atom/Feature/Common/Code/Source/Debug/RayTracingDebugFeatureProcessor.cpp

```diff
@@ -65,22 +70,47 @@ namespace AZ::Render
         m_settings.reset();
     }
 
+    void RayTracingDebugFeatureProcessor::OnRenderPipelineChanged(RPI::RenderPipeline* pipeline, [[maybe_unused]] RPI::SceneNotification::RenderPipelineChangeType changeType)
+    {
+        m_refreshPassEnabled = true;
+        m_pipeline = pipeline;
+    }
+
     void RayTracingDebugFeatureProcessor::AddRenderPasses(RPI::RenderPipeline* pipeline)
     {
         m_pipeline = pipeline;
+        if (m_debugComponentCount > 0)
+        {
+            auto pass = RPI::PassSystemInterface::Get()->CreatePassFromTemplate(Name{ "DebugRayTracingPassTemplate" }, Name{ "DebugRayTracingPass" });
+            if (!pass)
+            {
+                AZ_Assert(false, "Failed to create DebugRayTracingPass");
+                return;
+            }
+            m_pipeline->AddPassAfter(pass, Name{ "AuxGeomPass" });
+            m_refreshPassEnabled = true;
```

Comment:
Wouldn't it be more clean to call `pass->SetEnabled(m_settings->GetEnabled());` here and remove the usages of `m_refreshPassEnabled`?

Reply:
good point, updated it.